casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -98,8 +108,10 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
   ): F[Either[ReplayFailure, StateHash]] =
     withRuntimeLock { runtime =>
       for {
+        span   <- Metrics[F].span(replayComputeStateLabel)
         _      <- setBlockTime(blockTime, runtime)
-        result <- replayDeploys(startHash, terms, replayDeploy(runtime))
+        _      <- span.mark("before-replay-deploys")
+        result <- replayDeploys(startHash, terms, replayDeploy(runtime, span))
```

Comment:
passing the span to the called function feels wrong to me. I would have a Metrics effect with operations like: beginSpan, mark.
If you do that you won't have to pass the span and the functions will just place their own marks.

Just my 2c

Reply:
I prefer having the UI display all related spans in one table.

Note the above sentence may not be relevant.

Reply:
@odeac I'm not sure I understand what you are proposing. We don't want to keep in `Metrics` track of all spans. Our current implementation in `Metrics[F[_]]` is
```scala
def span(source: Metrics.Source): F[Span[F]]
```

Reply:
@KentShikama `RuntimeManager` is a `trait`. I don't want to introduce _spans_ into the traits methods parameter list. So `RuntimeManagerImpl` starts its own spans.